File: src/main/java/sharev/domain/account/repository/LinkRepository.java
package sharev.domain.account.repository;

import java.util.Collection;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;
import sharev.domain.account.entity.Link;

public interface LinkRepository extends JpaRepository<Link, Long> {

    List<Link> findAllByAccountIdIn(Collection<Long> accountIds);
}

File: src/main/kotlin/sharev/link/adapter/outbound/jpa/repository/LinkRepository.kt
package sharev.link.adapter.outbound.jpa.repository

import org.springframework.data.jpa.repository.JpaRepository
import sharev.link.adapter.outbound.jpa.entity.LinkJpaEntity

interface LinkRepository : JpaRepository<LinkJpaEntity, Long> {
    fun findAllByAccountIdIn(accountIds: Collection<Long>): List<LinkJpaEntity>

    fun findAllByAccountId(accountId: Long): List<LinkJpaEntity>
}
